superlesson/steps/annotate.py

```diff
@@ -16,58 +19,65 @@ def __init__(self, slides: Slides, lecture_notes: LessonFile):
-                        pypdf.generic.NumberObject(x),
-                        pypdf.generic.NumberObject(y),
-                        pypdf.generic.NumberObject(x + 30),
-                        pypdf.generic.NumberObject(y + 30),
-                    ]),
-                    pypdf.generic.NameObject("/Contents"): pypdf.generic.create_string_object(note_texts[index]),
-                    pypdf.generic.NameObject("/Open"): pypdf.generic.BooleanObject(True),
-                    pypdf.generic.NameObject("/Name"): pypdf.generic.NameObject("/Comment"),
-                })
-
-                # Add the note annotation to the page
-                if "/Annots" not in page:
-                    page[pypdf.generic.NameObject(
-                        "/Annots")] = pypdf.generic.ArrayObject()
-
-                page["/Annots"].append(note)
-
-            output_pdf.add_page(page)
-
-        output_path = self._lecture_notes.path / "transcription.pdf"
-        logging.info(f"Saving annotated pdf to {output_path}")
-        with open(output_path, "wb") as f:
-             output_pdf.write(f)
+        with open(self._lecture_notes.full_path, "rb") as a:
+            pdf = pypdf.PdfReader(a)
+            w = (
+                pdf.pages[0].mediabox.width / 72
+            )  # dividing by 72 to do pt to inch conversion
+            logging.debug(f"Value of 'w': {w}")
+            output = self._transcription_to_pdf(w)
+            with open(output, "rb") as t:
+                trans = pypdf.PdfReader(t)
+                merger = pypdf.PdfWriter()
+                for i in range(len(pdf.pages)):
+                    merger.append(fileobj=pdf, pages=(i, i + 1))
+                    if i < len(trans.pages):
+                        merger.append(fileobj=trans, pages=(i, i + 1))
+
+        with open(output, "wb") as f:
+            merger.write(f)
+        logging.info("PDF generation and merging completed.")
+
+    def _transcription_to_pdf(self, w) -> str:
+        preamble = f"""
+#set page(
+    width: {w}in,
+    height: auto,
+    margin: (
+        top: 2in,
+        bottom: 2in,
+        left: 1in,
+        right: 1in,
+    ),
+    fill: rgb("#fbfafa")
+)
+
+#set text(
+    size: 18pt,
+    hyphenate: false,
+    font: (
+        "Arial",
+    )
+)
+
+#set par(
+    justify: true,
+    leading: 1em,
+    first-line-indent: 0pt,
+    linebreaks: "optimized",
+)
+
+"""
+        with tempfile.NamedTemporaryFile(delete=False) as f:
+            f.write(preamble.encode("utf-8"))
+            f.write(
+                "\n#pagebreak()\n".join(
+                    [slide.transcription for slide in self.slides]
+                ).encode("utf-8")
+            )
+            temp_file_name = f.name
+
+        output = os.path.join(self._lecture_notes.path, "annotations.pdf")
+        typst.compile(temp_file_name, output=output)
+        logging.info("Transcription to PDF conversion completed.")
```

Comment:
I think this is also kinda confusing, you could do something like:

```suggestion
        logging.info(f"Intermediate PDF saved to: {output}")
```

And something similar on the other function.

Reply:
Done!